cashu/core/split.py

```diff
@@ -1,8 +1,7 @@
 def amount_split(amount):
     """Given an amount returns a list of amounts returned e.g. 13 is [1, 4, 8]."""
-    bits_amt = bin(amount)[::-1][:-2]
     rv = []
-    for (pos, bit) in enumerate(bits_amt):
-        if bit == "1":
-            rv.append(2**pos)
+    for i in range(amount.bit_length()):
```

Comment:
IMO should also add a check that `amount >= 0` and raise an `Exception` if not, as this behaves incorrectly if the `amount` is negative:

```python
In [1]: amount = -12

In [2]: rv = []
   ...: for i in range(amount.bit_length()):
   ...:     if amount & (1 << i):  # if bit i is set, add 2**i to list
   ...:         rv.append(1 << i)
   ...: 

In [3]: rv
Out[3]: [4]  # 4 != -12 ⁉️

In [4]: amount = 12

In [5]: rv = []
   ...: for i in range(amount.bit_length()):
   ...:     if amount & (1 << i):  # if bit i is set, add 2**i to list
   ...:         rv.append(1 << i)
   ...: 

In [6]: rv
Out[6]: [4, 8]  # 8 + 4 == 12 ✅
```



(Though the old version would also mess up with negative amounts as `bin(-12) == "-0b1100"` :rofl: so the `[:-2]` cut is wrong)

Reply:
Thanks, that makes sense. Added an `assert amount >= value` at the beginning of the function, i.e. an `AssertionError` exception is thrown if a negative value was passed.